admin/src/provideLoginState.js

```diff
@@ -1,16 +1,41 @@
+import 'babel-polyfill';
+import sg from 'sg.js';
+import { call } from 'sg.js/dist/effects';
 import { provideState, softUpdate } from 'freactal';
 
-export const submit = effects => state => effects.setToken(state.token).then(effects.setSecret(state.secret));
+import fetchEnvironments from './fetch/fetchEnvironments';
+
+export function* submit(effects, { projectId, token, secret }) {
+    try {
+        yield call(effects.setPending);
+        const environments = yield call(fetchEnvironments, { projectId, token });
+        yield call(effects.unsetPending);
+        yield call(effects.setConfig, {
+            projectId,
+            token,
+            secret,
+        });
+        yield call(effects.setEnvironments, environments);
+    } catch (error) {
+        yield call(effects.unsetPending);
+        yield call(effects.setError, error.message);
+    }
+}
 
 export const state = {
     initialState: () => ({
```

Comment:
Yes, but why ?
If you already have the token, secret and projectId then you won't instanciate this component.